Lib/site.py

```diff
@@ -509,32 +509,29 @@ def register_readline():
         pass
 
     if readline.get_current_history_length() == 0:
+        try:
+            from _pyrepl.main import CAN_USE_PYREPL
+        except ImportError:
+            CAN_USE_PYREPL = False
         # If no history was loaded, default to .python_history,
         # or PYTHON_HISTORY.
         # The guard is necessary to avoid doubling history size at
         # each interpreter exit when readline was already configured
         # through a PYTHONSTARTUP hook, see:
         # http://bugs.python.org/issue5845#msg198636
         history = gethistoryfile()
+        if os.getenv("PYTHON_BASIC_REPL") or not CAN_USE_PYREPL:
+            my_readline = readline
+        else:
+            my_readline = _pyrepl.readline
```

Comment:
Nitpick: I'm not a huge fan of `my_` variables, let's name it something else (you can bikeshed this):
```suggestion
        if os.getenv("PYTHON_BASIC_REPL") or not CAN_USE_PYREPL:
            target_readline = readline
        else:
            target_readline = _pyrepl.readline
```

Reply:
`readline_module` then

Reply:
> readline_module then

Yep, I like this.